compiler/rustc_macros/src/diagnostics/diagnostic_builder.rs

```diff
@@ -441,6 +437,11 @@ impl<'a> DiagnosticDeriveVariantBuilder<'a> {
                 code_field,
                 code_init,
             } => {
+                if let FieldInnerTy::Vec(_) = info.ty {
+                    throw_invalid_attr!(attr, &meta, |diag| diag
+                        .help("#[suggestion_*(...)] applied to `Vec` field is ambiguous"));
+                }
+
```

Comment:
I've improved the message. The problem is that `#[suggestion(code = "foo")] spans: Vec<Span>` could either mean:

```rust
for span in spans {
    span_suggestion(span, "foo");
}
```

or:

```rust
multipart_suggestion(spans.into_iter().map(|span| (span, "foo")));
```

In order to avoid ambiguity, a `Subdiagnostic` must be used (either a `Vec` of `#[suggestion]` subdiagnostics, or a single `#[multipart_suggestion]` subdiagnostic).

Reply:
Ah, so the change is not "forbid x" but "give a better error if x happens"? Thanks for this change, these sort of papercuts have been annoying me in the past 👍 

Please add this case to one of the files in https://github.com/rust-lang/rust/tree/master/src/test/ui-fulldeps/session-diagnostic

Reply:
It is "forbid x", but previously "x" was the first behaviour, which, while technically a correct interpretation, is probably not what most people would want. This behaviour wasn't actually used anywhere, it's just a preventative measure.

Reply:
Thanks for clarifying, that seems like a good idea.